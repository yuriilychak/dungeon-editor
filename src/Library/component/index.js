import React, {useCallback} from 'react';
import Fade from '@material-ui/core/Fade';
import {useDropzone} from 'react-dropzone';
import {makeStyles} from '@material-ui/styles';
import PropTypes from "prop-types";
import {useTranslation} from "react-i18next";

import ContentFolder from "./ContentFolder";
import TextureIcon from "../data/icon/texture.svg";
import SkeletonIcon from "../data/icon/skeleton.svg";
import ParticleIcon from "../data/icon/particle.svg";
import ElementIcon from "../data/icon/element.svg";
import FontIcon from "../data/icon/font.svg";

const useStyles = makeStyles({
    root: {
        width: "100%",
        height: "100%",
        maxHeight: "100%",
        margin: 0,
        position: "relative",
        boxShadow: "none",
        outline: "none"
    },
    overflow: {
        position: "absolute",
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        overflow: "auto"
    },
    drag: {
        border: "dashed #222222 2px",
        backgroundColor: "rgba(160,160,160,.5)",
        position: "absolute",
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        zIndex: 9999,
        opacity: 0,
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        pointerEvents: "none",
        '&:hover': {
            opacity: 1,
            transition: "opacity 2s ease-in"
        }
    }
});

const Library = props => {
    const {
        locales,
        tabs,
        files,
        onAddFiles,
        onRemoveElement,
        onRemoveParticle,
        onRemoveSkeletone,
        onRemoveFont,
        onRemoveTexture
    } = props;
    const {t} = useTranslation();
    const onDrop = useCallback(onAddFiles, []);
    const {getRootProps, getInputProps, isDragActive} = useDropzone({
        onDrop,
        noClick: true,
        noKeyboard: true
    });
    const classes = useStyles();

    const removeCallbacks = [
        onRemoveElement,
        onRemoveFont,
        onRemoveParticle,
        onRemoveSkeletone,
        onRemoveTexture
    ];

    const icons = [
        ElementIcon,
        FontIcon,
        ParticleIcon,
        SkeletonIcon,
        TextureIcon
    ];

    const emptyTabLocale = t(locales.emptyTab);
    const deleteItemLocale = t(locales.itemDelete);
    const renameItemLocale = t(locales.itemRename);

    const tabViews = tabs.map(tab => (
            <ContentFolder
                title={t(tab.locale)}
                key={tab.id}
                icon={icons[tab.id]}
                emptyText={emptyTabLocale}
                files={files[tab.id]}
                deleteText={deleteItemLocale}
                renameText={renameItemLocale}
                onRemoveElement={removeCallbacks[tab.id]}
            />
        )
    );

    return (
        <div {...getRootProps({className: 'dropzone'})} className={classes.root}>
            <input {...getInputProps()} />
            <div className={classes.overflow}>
                {tabViews}
            </div>
            <Fade in={isDragActive}>
                <div className={classes.drag}>
                    Drop the files here ...
                </div>
            </Fade>
        </div>
    );
};

Library.propTypes = {
    tabs: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number.isRequired,
        locale: PropTypes.string.isRequired
    })).isRequired,
    locales: PropTypes.shape({
        emptyTab: PropTypes.string.isRequired,
        itemDelete: PropTypes.string.isRequired,
        itemRename: PropTypes.string.isRequired
    }).isRequired,
    files: PropTypes.arrayOf(
        PropTypes.arrayOf(
            PropTypes.object.isRequired
        )
    )
};

export default Library;
