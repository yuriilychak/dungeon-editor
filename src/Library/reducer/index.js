import StaticData from "../data/index";
import STATE from "../state";

/**
 * @typedef {Object}
 * @name LibraryData
 * @property {number} id
 * @property {string} locale
 */

/**
 * @typedef {Object}
 * @name LibraryState
 * @property {Array<LibraryData>} tabs
 * @property {Array<Object>} files
 */

/**
 * @type {LibraryState}
 */

export const initialState = {
    ...StaticData,
    files: [[], [], [], [], []]
};

/**
 * @type {Object.<string, function(LibraryState, ActionData): LibraryState>}
 */

const actionHandlers = {
    [STATE.ADD_FILE]: (state, action) => {
        const files = state.files.slice(0);
        const {sectionId, data} = action.payload;
        const file = {
            title: data.name,
            id: data.id,
            hasPreview: false,
            isDirectory: false
        };
        files[sectionId] = [...files[sectionId], file];
        return {
            ...state,
            files
        }
    },
    [STATE.REMOVE_FILE]: (state, action) => {
        const files = state.files.slice(0);
        const index = action.payload.sectionId;
        files[index] = files[index].filter(file => file.id !== action.payload.id);
        return {
            ...state,
            files
        }
    },
    [STATE.UPDATE_TREE]: (state, action) => {
        const files = state.files.slice(0);
        const {fileTree, sectionId} = action.payload;
        files[sectionId] = fileTree;
        return {...state, files};
    },
    [STATE.ADD_DIRECTORY]: (state, action) => {
        const files = state.files.slice(0);
        const {sectionId, dirData} = action.payload;
        const fileTree = files[sectionId].slice(0);
        const parentId = dirData.parentId;
        const result = {
            title: dirData.name,
            id: dirData.id,
            hasPreview: false,
            isDirectory: true
        };

        if (parentId === -1) {
            fileTree.push(result);
        }
        else {
            const parent = findDir(fileTree, parentId);

            if (!parent) {
                return state;
            }

            if (!parent.children) {
                parent.children = [];
            }

            parent.children.push(result);
        }

        files[sectionId] = fileTree;

        return {...state, files};
    },
    [STATE.CLEAR]: state => ({...state, files: [[], [], [], [], []]})
};

function findDir(data, id) {
    const dirs = data.filter(element => element.isDirectory);
    let result = dirs.find( dir => dir.id === id);

    if (result) {
        return result;
    }

    dirs.forEach( dir => {
        if (!dir.children || result) {
            return;
        }

        result = findDir(dir.children, id);
    });

    return result;
}

/**
 * @param {LibraryState} state
 * @param {ActionData} action
 * @return {LibraryState}
 */

export default function topMenuReducer(state = initialState, action) {
    const actionHandler = actionHandlers[action.type];
    return actionHandler ? actionHandler(state, action) : state;
}
