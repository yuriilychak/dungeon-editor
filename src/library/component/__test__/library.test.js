import React from "react";
import 'babel-polyfill';

import { createMount } from '@material-ui/core/test-utils';

import { initialState } from '../../reducer';
import Library from "../library";

jest.mock("../../../common-ui/file-tree/file-tree");

describe('Library test',() => {

    const defaultProps = {
        ...initialState,
        files: [[], [], [], [], [], [], []],
        onAddDirectory: jest.fn(),
        onAddFiles: jest.fn(),
        onAddFile: jest.fn(),
        onDropFiles: jest.fn(),
        onRemoveFile: jest.fn(),
        onExportProject: jest.fn(),
        onPublishProject: jest.fn(),
        onRenameFile: jest.fn(),
        onSelectFile: jest.fn(),
        onUpdateTree: jest.fn(),
        onOpenFile: jest.fn()
    };

    it("default snapshot", () => {
        const wrapper = createMount()(
            <Library
                {...defaultProps}
            />
        );

        expect(wrapper.html()).toMatchSnapshot();
    });
});
