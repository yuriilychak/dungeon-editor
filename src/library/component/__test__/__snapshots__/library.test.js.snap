// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library test default snapshot 1`] = `"<div class=\\"titled-panel-root\\"><div class=\\"titled-panel-container\\"><span class=\\"MuiTypography-root titled-panel-text MuiTypography-button\\">Library</span><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add to library\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Export project\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 12v7H5v-7H3v7c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zm-6 .67l2.59-2.58L17 11.5l-5 5-5-5 1.41-1.41L11 12.67V3h2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Publish project\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path><path d=\\"M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"titled-panel-body\\" style=\\"padding: 0px;\\"><div class=\\"drop-area-root\\"><input multiple=\\"\\" type=\\"file\\" style=\\"display: none;\\" autocomplete=\\"off\\" tabindex=\\"-1\\"><div class=\\"drop-area-overflow\\"><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded\\"><div class=\\"MuiButtonBase-root MuiExpansionPanelSummary-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" aria-expanded=\\"false\\"><div class=\\"MuiExpansionPanelSummary-content\\"><img src=\\"undefined/icon/element_root.svg\\" style=\\"width: 16px; height: 16px;\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\"><div class=\\"library-section-header-title\\">Elements</div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new element\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new directory\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><path d=\\"M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3v2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"false\\" aria-hidden=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></div></div><div class=\\"MuiCollapse-container MuiCollapse-hidden\\" style=\\"min-height: 0px;\\"><div class=\\"MuiCollapse-wrapper\\"><div class=\\"MuiCollapse-wrapperInner\\"><div role=\\"region\\"><div class=\\"MuiExpansionPanelDetails-root\\"><span class=\\"section-body-empty\\">Category empty. Add files via drag'n'drop or add icon</span></div></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded\\"><div class=\\"MuiButtonBase-root MuiExpansionPanelSummary-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" aria-expanded=\\"false\\"><div class=\\"MuiExpansionPanelSummary-content\\"><img src=\\"undefined/icon/font_root.svg\\" style=\\"width: 16px; height: 16px;\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\"><div class=\\"library-section-header-title\\">Fonts</div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new element\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new directory\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><path d=\\"M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3v2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"false\\" aria-hidden=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></div></div><div class=\\"MuiCollapse-container MuiCollapse-hidden\\" style=\\"min-height: 0px;\\"><div class=\\"MuiCollapse-wrapper\\"><div class=\\"MuiCollapse-wrapperInner\\"><div role=\\"region\\"><div class=\\"MuiExpansionPanelDetails-root\\"><span class=\\"section-body-empty\\">Category empty. Add files via drag'n'drop or add icon</span></div></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded\\"><div class=\\"MuiButtonBase-root MuiExpansionPanelSummary-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" aria-expanded=\\"false\\"><div class=\\"MuiExpansionPanelSummary-content\\"><img src=\\"undefined/icon/particle_root.svg\\" style=\\"width: 16px; height: 16px;\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\"><div class=\\"library-section-header-title\\">Particles</div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new element\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new directory\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><path d=\\"M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3v2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"false\\" aria-hidden=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></div></div><div class=\\"MuiCollapse-container MuiCollapse-hidden\\" style=\\"min-height: 0px;\\"><div class=\\"MuiCollapse-wrapper\\"><div class=\\"MuiCollapse-wrapperInner\\"><div role=\\"region\\"><div class=\\"MuiExpansionPanelDetails-root\\"><span class=\\"section-body-empty\\">Category empty. Add files via drag'n'drop or add icon</span></div></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded\\"><div class=\\"MuiButtonBase-root MuiExpansionPanelSummary-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" aria-expanded=\\"false\\"><div class=\\"MuiExpansionPanelSummary-content\\"><img src=\\"undefined/icon/skeleton_root.svg\\" style=\\"width: 16px; height: 16px;\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\"><div class=\\"library-section-header-title\\">Skeletons</div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new element\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new directory\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><path d=\\"M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3v2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"false\\" aria-hidden=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></div></div><div class=\\"MuiCollapse-container MuiCollapse-hidden\\" style=\\"min-height: 0px;\\"><div class=\\"MuiCollapse-wrapper\\"><div class=\\"MuiCollapse-wrapperInner\\"><div role=\\"region\\"><div class=\\"MuiExpansionPanelDetails-root\\"><span class=\\"section-body-empty\\">Category empty. Add files via drag'n'drop or add icon</span></div></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded\\"><div class=\\"MuiButtonBase-root MuiExpansionPanelSummary-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" aria-expanded=\\"false\\"><div class=\\"MuiExpansionPanelSummary-content\\"><img src=\\"undefined/icon/sound_root.svg\\" style=\\"width: 16px; height: 16px;\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\"><div class=\\"library-section-header-title\\">Sounds</div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new element\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new directory\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><path d=\\"M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3v2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"false\\" aria-hidden=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></div></div><div class=\\"MuiCollapse-container MuiCollapse-hidden\\" style=\\"min-height: 0px;\\"><div class=\\"MuiCollapse-wrapper\\"><div class=\\"MuiCollapse-wrapperInner\\"><div role=\\"region\\"><div class=\\"MuiExpansionPanelDetails-root\\"><span class=\\"section-body-empty\\">Category empty. Add files via drag'n'drop or add icon</span></div></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded\\"><div class=\\"MuiButtonBase-root MuiExpansionPanelSummary-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" aria-expanded=\\"false\\"><div class=\\"MuiExpansionPanelSummary-content\\"><img src=\\"undefined/icon/texture_root.svg\\" style=\\"width: 16px; height: 16px;\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\"><div class=\\"library-section-header-title\\">Textures</div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new element\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new directory\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><path d=\\"M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3v2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"false\\" aria-hidden=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></div></div><div class=\\"MuiCollapse-container MuiCollapse-hidden\\" style=\\"min-height: 0px;\\"><div class=\\"MuiCollapse-wrapper\\"><div class=\\"MuiCollapse-wrapperInner\\"><div role=\\"region\\"><div class=\\"MuiExpansionPanelDetails-root\\"><span class=\\"section-body-empty\\">Category empty. Add files via drag'n'drop or add icon</span></div></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded\\"><div class=\\"MuiButtonBase-root MuiExpansionPanelSummary-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" aria-expanded=\\"false\\"><div class=\\"MuiExpansionPanelSummary-content\\"><img src=\\"undefined/icon/tile_map_root.svg\\" style=\\"width: 16px; height: 16px;\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\"><div class=\\"library-section-header-title\\">Tile maps</div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new element\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"Add new directory\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><path d=\\"M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3v2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"false\\" aria-hidden=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></div></div><div class=\\"MuiCollapse-container MuiCollapse-hidden\\" style=\\"min-height: 0px;\\"><div class=\\"MuiCollapse-wrapper\\"><div class=\\"MuiCollapse-wrapperInner\\"><div role=\\"region\\"><div class=\\"MuiExpansionPanelDetails-root\\"><span class=\\"section-body-empty\\">Category empty. Add files via drag'n'drop or add icon</span></div></div></div></div></div></div></div><div class=\\"drop-area-drag\\" style=\\"opacity: 0; visibility: hidden;\\">Drop the files here ...</div></div></div></div>"`;
